Sources/MMIO/Documentation.docc/Getting-Started/Defining-Registers.md

```diff
@@ -0,0 +1,102 @@
+# Defining Registers
+
+Learn how to describe hardware register layouts using Swift MMIO's powerful macros.
+
+## Overview
+
+Swift MMIO employs a declarative approach to define the structure of memory-mapped peripherals and their registers. Instead of manual pointer arithmetic and bit manipulation, you use Swift macros to create type-safe and readable hardware interfaces.
+
+The core macros for this purpose are:
+- ``MMIO/RegisterBlock()``: Groups registers into a peripheral or a logical block.
+- ``MMIO/Register(bitWidth:)``: Defines the bit-level layout of an individual hardware register.
+- Bit Field Macros (e.g., ``MMIO/ReadWrite(bits:as:)``, ``MMIO/ReadOnly(bits:as:)``): Specify individual named fields within a register.
+
+## Automating Definitions with SVD2Swift
```

Comment:
Can we have steps how to use SVD2Swift, and/or an example use?

Reply:
Yeah I should link to: https://swiftpackageindex.com/apple/swift-mmio/main/documentation/svd2swift